desktop-exporter/app/components/waterfall-view/waterfall-row.tsx

```diff
@@ -25,40 +25,47 @@ export function WaterfallRow({ index, style, data }: WaterfallRowProps) {
     selectedSpanID,
     setSelectedSpanID,
   } = data;
-  let { spanID, spanData } = orderedSpans[index];
-  let spanDepth = orderedSpans[index].metadata.depth;
+
+  let span = orderedSpans[index];
+  let { spanID, depth } = span.metadata;
 
   // Set the background colour to make the list striped.
   let backgroundColour =
     index % 2 ? "" : useColorModeValue("gray.50", "gray.700");
+  let missingColour = useColorModeValue("orange.100", "orange.700");
   let selectedColour = useColorModeValue("pink.50", "pink.900");
 
-  //Set the style for the selected item
-  if (!!selectedSpanID && selectedSpanID === spanID) {
-    backgroundColour = selectedColour;
-  }
-
   // Set the padding to indicate parent/children relationship between spans
-  let paddingLeft = spanDepth ? spanDepth * 25 : 0;
+  let paddingLeft = depth * 25;
 
-  // Add zero-width space after forward slashes, dashes, and dots
-  // to indicate line breaking opportunity
-  let nameLabel = spanData
-    ? spanData.name
-        .replaceAll("/", "/\u200B")
-        .replaceAll("-", "-\u200B")
-        .replaceAll(".", ".\u200B")
-    : "missing span";
+  let nameLabel;
+  let resourceLabel;
+
+  if (span.status) {
```

Comment:
I was a little confused how this is working since `span.status` is not nullable, but I think this is currently relying on `SpanDataStatus.missing` being `0`. If we change it to string values (see other comment), this will break, so let's do an explicit check here:

```typescript
if (span.status === SpanDataStatus.present) {
```